firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/BreathingCircuit/AlarmMuteService.cpp

```diff
@@ -11,9 +11,24 @@
 
 namespace Pufferfish::Driver::BreathingCircuit {
 
+void AlarmMuteService::input_clock(uint32_t current_time) {
+  if (mute_start_time_ == 0) {
+    mute_start_time_ = current_time;
+  }
+  mute_duration_ = current_time - mute_start_time_;
+}
+
 void AlarmMuteService::transform(
-    AlarmMute &alarm_mute, const AlarmMuteRequest &alarm_mute_request) {
+    uint32_t current_time, const AlarmMuteRequest &alarm_mute_request, AlarmMute &alarm_mute) {
   alarm_mute.active = alarm_mute_request.active;
```

Comment:
I just remembered...`AlarmMuteService` should automatically set `alarm_mute.active` to `false` once `duration >= mute_max_duration` is satisfied; even if `alarm_mute_request.active == true`, `AlarmMuteService` should only set `alarm_mute.active = alarm_mute_request.active` after a timeout, and/or only when `alarm_mute_request` has a new starting time which is past the ending time of the previous alarm mute. If this is too complicated for this PR, or if this depends on #372, let's fix this in a future PR.